### PR TITLE
SER-1517 component created for loan allocation products

### PR DESCRIPTION
**Pull request by SauravPriyadarshy**
The whole loan allocation protocol involves the list view, the add view as currently implemented and visible in loans.test. Implemented all screens for the loan allocation process.

## Related issues and discussion
#{Issue Number}

## Screenshots, if any
image
image

### CODE REVIEW

File: src/app/products/loan-product-allocation-setting/loan-product-allocation-setting.component.html [hunk 1]
```diff
@@ -0,0 +1,52 @@
+<div class="container m-b-20" fxLayout="row" fxLayoutAlign="end" fxLayoutGap="20px">
+    <button mat-raised-button color="primary" [routerLink]="['create']" *mifosxHasPermission="'CREATE_LOANPRODUCT'">
+      <fa-icon icon="plus"></fa-icon>&nbsp;&nbsp;
```
Comment (william000000) [inline src/app/products/loan-product-allocation-setting/loan-product-allocation-setting.component.html:3]: The permission should be `CREATE_LOANPAYMENTALLOCATIONSETTING`

**Review by william000000 — CHANGES_REQUESTED** [review]

File: src/app/products/products.component.html [hunk 2]
```diff
@@ -12,9 +12,16 @@
             <mat-icon matListIcon>
               <fa-icon icon="briefcase" size="sm"></fa-icon>
             </mat-icon>
-            <h4 matLine>Loan Products</h4>
+            <h4 matLine>Loan payment</h4>
             <p matLine>Add new loan product or modify or inactivate loan product</p>
```
Comment (william000000) [inline src/app/products/products.component.html:16]: I think the title was already correct

File: src/app/products/products.component.html [hunk 3]
```diff
@@ -12,9 +12,16 @@
...
           </mat-list-item>
+          <mat-list-item [routerLink]="['/loanproductallocationsettings']" *mifosxHasPermission="'READ_LOANPRODUCT'">
+            <mat-icon matListIcon>
```
Comment (william000000) [inline src/app/products/products.component.html:19]: And it would be `READ_LOANPAYMENTALLOCATIONSETTING`

**Review by gilclefes — APPROVED** [review]
looks good to me.

**Review by william000000 — APPROVED** [review]
Lgtm

**Pull request merged**